### PR TITLE
Feat js t2002

### PR DESCRIPTION
**Pull request by sganeshkumaar**

### CODE REVIEW

**Review by lakshmipooja1510 — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 1]
```diff
@@ -15,7 +19,11 @@ async function loadFunction () {
 
   createCityNames()
 
-  fillContentOnLoad()
+  displayedCards = cityNames
+
+  await fillContentOnLoad()
+
+  console.log('ready to roll')
```
[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/index.js:25]: all consoles were removed

File: src/index.js [hunk 2]
```diff
@@ -250,15 +261,297 @@ function fillTopOnInvalid () {
...
+    for (let i = 0; i < 2; i++) {
+      arrows[i].style.display = 'inline'
+    }
+  } else {
+    cardsContainer.style.justifyContent = 'center'
+    for (let i = 0; i < 2; i++) {
+      arrows[i].style.display = 'none'
+    }
+  }
+  console.log()
```
[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/index.js:476]: ill try to remove all consoles before raising upcoming PRs

File: src/index.js [hunk 3]
```diff
@@ -250,15 +261,297 @@ function fillTopOnInvalid () {
...
+    if (!(temperature > 29)) return false
+
+    if (!(humidity < 50)) return false
+
+    if (!(precipitation >= 50)) return false
+
+    return true
+  }
+
+  if (prefernce === 'snowflake') {
```
[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/index.js:405]: this spelling has been changed

File: src/index.js [hunk 4]
```diff
@@ -250,15 +261,297 @@ function fillTopOnInvalid () {
...
+    fillContentTop(city)
   }
-  fillContentTop(city)
+}
+
+/**
+ * @param {object} input object of the spinner input tag
+ * actions that should happen when spinner input change
+ */
+function actionOnSpinnerChnage (input) {
```
[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/index.js:496]: changed this spelling

**Comment by sganeshkumaar** [comment]
modified timer to run

File: src/index.js [hunk 5]
```diff
@@ -250,15 +284,352 @@ function fillTopOnInvalid () {
...
+/**
+ * @param {object} icon object of the selected icon
+ * actions to be done when a particular preference is selected
+ */
+function actionOnPreferenceClick (icon) {
+  const preference = icon.target.id.slice(11)
+
+  changePreferenceIndicator(preference)
+
+  const filteredCities = cityNames.filter(filterOnPrefernece, preference)
```
Comment (lakshmipooja1510) [inline src/index.js:389]: filterOnPrefernece -> check spelling

File: src/index.js [hunk 6]
```diff
@@ -250,15 +284,352 @@ function fillTopOnInvalid () {
...
+  }
+  return divisionNumber
+}
+
+/**
+ * @param {string} city name of the card city
+ * @returns {boolean} wheather if a city belongs to a selected preference or not
+ */
+function filterOnPrefernece (city) {
+  const preference = String(this)
```
Comment (lakshmipooja1510) [inline src/index.js:429]: check filterOnPrefernece  spelling

**Review by lakshmipooja1510 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 5 above)
Comment (sganeshkumaar) [reply src/index.js]: Spelling of the function changed

**Review by lakshmipooja1510 — APPROVED** [review]

**Pull request merged**